torch/testing/_internal/common_device_type.py

```diff
@@ -788,6 +788,72 @@ def test_wrapper(*args, **kwargs):
                     print("Failed to instantiate {0} for op {1}!".format(test_name, op.name))
                     raise ex
 
+# Decorator that generates two set of tests with layout=torch.strided and layout=specified_layout.
+# All keyword arguments are used only for the specified layout, so they have no effect on the original test.
+class enableLayout(_TestParametrizer):
+    def __init__(self, layout: torch.layout, *,
```

Comment:
1. Would it make sense to allow an arbitrary number of layout?  Meaning I could do `@enableLayout(torch.sparse_coo, sparse_scr)` to get 3 tests.
2. If we do 1., does it make sense to always use strided or are there use cases where I for example want sparse COO and CSR but no strided tensors?

Reply:
1. Yes, I'd modify the code later when such use case arises. Not doing it now because the dtype, device, and operator coverage is not the same for CSR and COO matrices, so separate dtypes and device skips would be needed for each layout.

I think it would better to have in the future the possibility to compose the decorators
```python
@enableLayout(torch.sparse_coo, dtypes=..., )
@enableLayout(torch.sparse_csr, dtypes=..., )
```
to get 3 tests.
2. There would only be only a few functions specific to the sparse layout, so I think using always strided as default layout is fine.